lib/ansible/modules/cloud/amazon/ecs_attribute.py

```diff
@@ -0,0 +1,311 @@
+version_added: "2.4"
+author: Andrej Svenke (@anryko)
+requirements: [ botocore, boto3 ]
+options:
+    cluster:
+        description:
+            - The short name or full Amazon Resource Name (ARN) of the cluster
+              that contains the resource to apply attributes.
+        required: true
+    state:
+        description:
+            - The desired state of the attributes.
+        required: false
+        default: present
+        choices: ['present', 'absent']
+    attributes:
+        description:
+            - List of attributes.
+        required: true
+        suboptions:
+            name:
+                description:
+                    - The name of the attribute. Up to 128 letters (uppercase and lowercase),
+                      numbers, hyphens, underscores, and periods are allowed.
+                required: true
+            value:
+                description:
+                    - The value of the attribute. Up to 128 letters (uppercase and lowercase),
+                      numbers, hyphens, underscores, periods, at signs (@), forward slashes, colons,
+                      and spaces are allowed.
+                required: false
+    ec2_instance_id:
+        description:
+            - EC2 instance ID of ECS cluster container instance.
+        required: true
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Set attributes
+- ecs_attribute:
+    state: present
+    cluster: test-cluster
+    ec2_instance_id: "{{ ec2_id }}"
+    attributes:
+      - flavor: test
+      - migrated
+  delegate_to: localhost
+
+# Delete attributes
+- ecs_attribute:
+    state: absent
+    cluster: test-cluster
+    ec2_instance_id: "{{ ec2_id }}"
+    attributes:
+      - flavor: test
+      - migrated
+  delegate_to: localhost
+'''
+
+RETURN = '''
+attributes:
+    description: attributes
+    type: complex
+    returned: always
+    contains:
+        cluster:
+            description: cluster name
+            type: string
+        ec2_instance_id:
+            description: ec2 instance id of ecs container instance
+            type: string
+        attributes:
+            description: list of attributes
+            type: list of complex
+            contains:
+                name:
+                    description: name of the attribute
+                    type: string
+                value:
+                    description: value of the attribute
+                    returned: if present
+                    type: string
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, EndpointConnectionError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsAttributes:
```

Comment:
Need to use new-style Python 2 classes, as this Py3 class declaration won't behave right in Python 2 environments. Change to `class EcsAttributes(object):`